### ISSUE TITLE
(🐞) Wrong error message on overloaded function with unexpected keyword parameter

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#11927]
`No overload variant of "foo" matches argument type "int"`

This error message is flat out incorrect as there is a variant that matches `int` but there is an unexpected parameter name.

```py
from typing import overload

@overload
def foo(a: int) -> None: ...

@overload
def foo(a: str) -> None: ...

def foo(a: object) -> None: ...

def bar(a: int) -> None: ...

foo(b=1)
bar(b=1)
```
```
main.py:11: note: "bar" defined here
main.py:13: error: No overload variant of "foo" matches argument type "int"  [call-overload]
main.py:13: note: Possible overload variants:
main.py:13: note:     def foo(a: int) -> None
main.py:13: note:     def foo(a: str) -> None
main.py:14: error: Unexpected keyword argument "b" for "bar"  [call-arg]
Found 2 errors in 1 file (checked 1 source file)
```

[playground link](https://mypy-play.net/?mypy=master&python=3.10&flags=show-error-codes%2Cstrict&gist=ed12784a53d28ae7acb8c7864f8fd21e)

# Also
In that output, for what reason is there a random `main.py:11: note: "bar" defined here`? Should I raise that as a separate issue?

**Comment by jbbqqf** [comment]
Hi! Triaging older issues — I think this one can be closed: the error message has been improved upstream and now points clearly at the unexpected keyword argument.

Evidence:

- The fix landed in [#20592 "Improve error messages for unexpected keyword arguments in overloaded functions"](https://github.com/python/mypy/pull/20592), merged 2026-04-17 (commit [`3c3065c`](https://github.com/python/mypy/commit/3c3065cb106e22205365b7f0e74b6c77c95bf4ae)).
- Reproducing the original snippet from the issue with current `master` gives:
  ```
  test.py:13: error: Unexpected keyword argument "b" for overloaded function "foo"  [call-overload]
  test.py:14: error: Unexpected keyword argument "b" for "bar"  [call-arg]
  ```
  i.e. the message is now specifically "Unexpected keyword argument" rather than the misleading "No overload variant of \"foo\" matches argument type \"int\"".
- The new behavior is implemented in [`mypy/messages.py`](https://github.com/python/mypy/blob/master/mypy/messages.py) (via `unexpected_keyword_argument*` helpers) and exercised by [`test-data/unit/check-overloading.test`](https://github.com/python/mypy/blob/master/test-data/unit/check-overloading.test) (new `testOverloadKwargUnexpected` cases added in #20592).

Would you mind closing this one out? Happy to be wrong — if you still see the old wording on a current build, please share a repro and we can dig in.

---

🤖 Disclosure: I'm a data engineer running triage on older issues with help from an AI assistant (Claude). I verified the repro and citations locally on `master` before posting. If anything here is off, please flag it and I'll correct.
